src/utils/buffer.ts

```diff
@@ -12,3 +14,134 @@ export const hexToBuffer = (value: string): Buffer => {
   }
   return Buffer.from(value, 'hex');
 };
+
+/**
+ * Unpacks a buffer size
+ *
+ * @param {number} n The size of the buffer to unpack
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[Buffer, Buffer]} The unpacked buffer followed by the rest of the buffer
+ */
+export const unpackLen = (n: number, buff: Buffer): [Buffer, Buffer] => {
+  return [
+    buff.slice(0, n),
+    buff.slice(n)
+  ];
+};
+
+/**
+ * Unpacks an integer from a buffer
+ *
+ * @param {number} n The size of the number in bytes
+ * @param {boolean} signed If the number is signed
+ * @param {Buffer} buff The buffer to unpack
+ *
+ * @return {[number, Buffer]} The unpacked number followed by the rest of the buffer
+ */
+export const unpackToInt = (n: number, signed: boolean, buff: Buffer): [number, Buffer] => {
+  let retInt;
+  const slicedBuff = buff.slice(0, n);
+  if (n === 1) {
+    if (signed) {
+      retInt = slicedBuff.readInt8();
+    } else {
+      retInt = slicedBuff.readUInt8();
+    }
+  } else if (n === 2) {
+    if (signed) {
+      retInt = slicedBuff.readInt16BE();
+    } else {
+      retInt = slicedBuff.readUInt16BE();
+    }
+  } else if (n === 4) {
+    if (signed) {
+      retInt = slicedBuff.readInt32BE();
+    } else {
+      retInt = slicedBuff.readUInt32BE();
+    }
+  } else if (n === 8) {
+    // We have only signed ints here
+    if (parseInt(nodeMajorVersion) > 8) {
+      retInt = Number(slicedBuff.readBigInt64BE());
+    } else {
+      retInt = slicedBuff.readIntBE(0, 8);
+    }
+
+  }
```

Comment:
Add `} else { throw "invalid value for n"; }`.

Reply:
Done